### PR TITLE
VP-5059

### PR DESCRIPTION
**Pull request by krankenbro**

### CODE REVIEW

**Review by dimitri-ak — APPROVED** [review]

File: add-version-suffix/index.js [hunk 1]
```diff
@@ -42,16 +42,14 @@ async function getCommitCount(baseBranch) {
...
-    {
-        const commitCount = await getCommitCount(branchName);
-        await exec.exec(`vc-build ChangeVersion -CustomVersionSuffix \"alpha.${commitCount}\"`).then(exitCode => {
-            if(exitCode != 0)
-            {
-                core.setFailed("vc-build ChangeVersion failed");
-            }
-        });
-    }
+    const commitCount = await getCommitCount(branchName);
```
Comment (mvktsk) [inline add-version-suffix/index.js:45]: Can we add versionSuffix input parameter and use get-image-version.suffix value?

File: build-theme/action.yml [hunk 2]
```diff
@@ -1,5 +1,14 @@
 name: 'build-theme'
 description: 'Builds theme'
+inputs:
+   release_branch: 
+    description: "Branch support preparation of a new production release"
+    required: false
+    default: "master"
+   versionSuffix:
```
Comment (mvktsk) [inline build-theme/action.yml:8]: Rename `release_branch` to `releaseBranch`

File: deploy-workflow/workflows/modules/.github/workflows/main.yml [hunk 3]
```diff
@@ -60,15 +60,38 @@ jobs:
...
+        uses: VirtoCommerce/vc-github-actions/sonar-quality-gate@master
+        with:
+          login: ${{secrets.SONAR_TOKEN}}
+
       - name: Packaging
         run: vc-build Compress -skip Clean+Restore+Compile+Test
       
-      # Step runs only on commit to master or dev brunch
       - name: Publish Nuget
         uses: VirtoCommerce/vc-github-actions/publish-nuget@master
```
Comment (mvktsk) [inline deploy-workflow/workflows/modules/.github/workflows/main.yml:72]: Add check `if: ${{ github.ref == 'refs/heads/dev' || github.ref == 'refs/heads/master'}}`
In VirtoCommerce/vc-github-actions/publish-nuget remove check `if(branchName === 'dev' || branchName === 'master')`

File: deploy-workflow/workflows/platform/.github/workflows/main.yml [hunk 4]
```diff
@@ -65,36 +65,37 @@ jobs:
...
-      with:
-        githubUser: ${{ env.GITHUB_ACTOR }}
-        githubToken: ${{ env.GITHUB_TOKEN }}
-        dockerTag: dev-linux-latest
-        platformImage: docker.pkg.github.com/virtocommerce/vc-platform/platform
-        storefrontImage: docker.pkg.github.com/virtocommerce/vc-storefront/storefront
           
-    # Step runs only on commit to master or dev brunch
     - name: Publish Nuget
       uses: VirtoCommerce/vc-github-actions/publish-nuget@master
```
Comment (mvktsk) [inline deploy-workflow/workflows/platform/.github/workflows/main.yml:84]: Add check if: ${{ github.ref == 'refs/heads/dev' || github.ref == 'refs/heads/master'}}

File: deploy-workflow/workflows/storefront/.github/workflows/main.yml [hunk 5]
```diff
@@ -64,24 +64,34 @@ jobs:
...
     - name: Packaging
       run: vc-build Compress -skip Clean+Restore+Compile+Test
 
-    # Step runs only on commit to master or dev brunch
     - name: Build Docker Image
       if: ${{ github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev' }}
       id: dockerBuild
       uses: VirtoCommerce/vc-github-actions/build-docker-image@master
       with:
         tag: ${{ steps.image.outputs.taggedVersion }}
```
Comment (mvktsk) [inline deploy-workflow/workflows/storefront/.github/workflows/main.yml:80]: Add parameters 
```
        imageName: "storefront"
        dockerFiles: "https://raw.githubusercontent.com/VirtoCommerce/vc-docker/master/linux/storefront/Dockerfile"
```

File: deploy-workflow/workflows/platform/.github/workflows/main.yml [hunk 6]
```diff
@@ -65,36 +65,37 @@ jobs:
...
+
     - name: Packaging
       run: vc-build Compress -skip Clean+Restore+Compile+Test
 
     - name: Build Docker Image
-      if: ${{ github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev' || github.ref == 'refs/heads/dev-workflow' }}
+      if: ${{ github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev' }}
       id: dockerBuild
       uses: VirtoCommerce/vc-github-actions/build-docker-image@master
       with:
```
Comment (mvktsk) [inline deploy-workflow/workflows/platform/.github/workflows/main.yml:80]: Add parameters
```
        imageName: "platform"
        dockerFiles: "https://raw.githubusercontent.com/VirtoCommerce/vc-docker/master/linux/platform/Dockerfile;https://raw.githubusercontent.com/VirtoCommerce/vc-docker/master/linux/platform/wait-for-it.sh"
```

File: deploy-workflow/workflows/themes/.github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      if: ${{ github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev' }}
+      id: publish
+      uses: VirtoCommerce/vc-github-actions/publish-theme@master
+
+    - name: Create Release
+      if: github.ref == 'refs/heads/master'
+      id: create_release
+      uses: actions/create-release@v1
+      with:
+        tag_name: ${{ steps.image.outputs.tag }}
```
Comment (mvktsk) [inline deploy-workflow/workflows/themes/.github/workflows/main.yml:65]: change `steps.image.outputs.tag` to s`teps.image.outputs.prefix`

File: deploy-workflow/workflows/themes/.github/workflows/main.yml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        release_name: ${{ steps.image.outputs.tag }}
```
Comment (mvktsk) [inline deploy-workflow/workflows/themes/.github/workflows/main.yml:66]: change `steps.image.outputs.tag` to s`teps.image.outputs.prefix`

File: get-image-version/action.yml [hunk 9]
```diff
@@ -1,28 +1,35 @@
 name: 'Get Image Version'
-description: 'This action grabs Version and Version suffix from module.manifest, package.json or Directory.Build.Props file. If Version suffix not present calculate it as a branch commits count'
+description: 'This action grabs Version and Version suffix from module.manifest, package.json or Directory.Build.props file. If Version suffix not present calculate it as a branch commits count'
 inputs:
   path:
-    description: 'Path to directory that contains module.manifest, package.json or Directory.Build.Props'
-    required: false    
+    description: 'Path to directory that contains module.manifest, package.json or Directory.Build.props'
+    required: false
+  release_branch:
```
Comment (mvktsk) [inline get-image-version/action.yml:7]: My mistake, please rename `release_branch` to `releaseBranch`

File: get-image-version/index.js [hunk 10]
```diff
@@ -208,15 +205,16 @@ let branchName = "";
 async function run() 
 {
     // let files = findFile("src", "module.manifest");
+    const releaseBranch = core.getInput("release_branch");
```
Comment (mvktsk) [inline get-image-version/index.js:208]: Rename `release_branch` to `releaseBranch`

File: publish-blob-release/src/index.ts [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+        if(exitCode != 0)
+        {
+            core.setFailed("azcopy failed");
+            process.exit(exitCode);
+        }
+    });
+    core.setOutput("packageUrl", downloadUrl);
+}
+
+// async function run() {
```
Comment (mvktsk) [inline publish-blob-release/src/index.ts:44]: Remove commented code

File: publish-docker-image/index.js [hunk 12]
```diff
@@ -17,14 +17,19 @@ async function changeTag(image, oldTag, newTag)
 async function renameImage(oldName, oldTag, newName, newTag)
 {
     let command = `docker tag ${oldName}:${oldTag} ${newName}:${newTag}`;
-    await exec.exec(command);
+    await exec.exec (command);
```
Comment (mvktsk) [inline publish-docker-image/index.js:20]: Remove blank between exec and (command)

File: publish-docker-image/index.js [hunk 13]
```diff
@@ -33,36 +38,41 @@ async function run()
     let branchName = await utils.getBranchName(github);
     const imageName = core.getInput("image");
     const tag = core.getInput("tag");
-    const dockerUser = core.getInput("docker_user");
-    const dockerToken = core.getInput("docker_token");
-    const dockerHub = core.getInput("docker_hub");
+    const dockerUser = core.getInput("docker_user").toLocaleLowerCase();
```
Comment (mvktsk) [inline publish-docker-image/index.js:41]: My mistake  please remove `.toLocaleLowerCase()`

File: publish-docker-image/index.js [hunk 14]
```diff
@@ -33,36 +38,41 @@ async function run()
...
+    const dockerToken = core.getInput("docker_token").toLocaleLowerCase();
```
Comment (mvktsk) [inline publish-docker-image/index.js:42]: My mistake, please remove `.toLocaleLowerCase()`

File: publish-docker-image/index.js [hunk 15]
```diff
@@ -33,36 +38,41 @@ async function run()
...
+    const dockerHub = core.getInput("docker_hub").toLocaleLowerCase();
```
Comment (mvktsk) [inline publish-docker-image/index.js:43]: My mistake, please remove `.toLocaleLowerCase()`

File: publish-docker-image/index.js [hunk 16]
```diff
@@ -33,36 +38,41 @@ async function run()
...
+    const releaseBranch = core.getInput("release_branch").toLocaleLowerCase();
```
Comment (mvktsk) [inline publish-docker-image/index.js:44]: My mistake, please remove `.toLocaleLowerCase()`

File: build-theme/index.js [hunk 17]
```diff
@@ -51,21 +18,18 @@ async function buildTheme()
 
 async function run()
 {
+    const releaseBranch = core.getInput("release_branch");
```
Comment (mvktsk) [inline build-theme/index.js:21]: Rename `release_branch` to `releaseBranch`

**Review by mvktsk — CHANGES_REQUESTED** [review]

File: build-theme/action.yml [hunk 18]
```diff
@@ -1,5 +1,14 @@
 name: 'build-theme'
 description: 'Builds theme'
+inputs:
+   release_branch:
```
Comment (mvktsk) [inline build-theme/action.yml:4]: Rename `release_branch` to `releaseBranch`

**Review by mvktsk — CHANGES_REQUESTED** [review]

**Review by dimitri-ak — APPROVED** [review]

**Review by mvktsk — APPROVED** [review]

**Pull request merged**